### PR TITLE
Appended an _id suffix to all related field names in the rest api

### LINKED ISSUE #1597: Rename ForeignKey fields to FIELDNAME_id
Please rename the field names of foreign keys, that only return the id to FIELDNAME_id.

For example the motion serializer has the field 'category' which is the id of the category.

On the client, I would like to add the category-object to the motion object, so motion.category.name returns the name of the category. But my problem is, that the attribute-name 'category' is already taken.

So it should be the same then in django. FIELD_id is the id of a ForeignKeyField and FIELD is the foreign object.

This is a blocker for #1596

### PR DESCRIPTION
**Pull request by ostcar**
Fixes #1597

This pull requests breaks the angular client in some parts. I will fix it in #1596

### CODE REVIEW

**Comment by ostcar** [comment]
This is a blocker for #1596

**Comment by ostcar** [comment]
Please wait with your review. I have to test something.

**Comment by ostcar** [comment]
@normanjaeckel please review.

Also see https://github.com/tomchristie/django-rest-framework/issues/3121 and https://github.com/tomchristie/django-rest-framework/pull/3122

File: openslides/utils/rest_api.py [hunk 1]
```diff
@@ -30,6 +32,69 @@
 router = DefaultRouter()
 
 
+class IdManyRelatedField(ManyRelatedField):
+    field_name_suffix = '_id'
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: Can you please a a docstring here?

File: openslides/utils/rest_api.py [hunk 2]
```diff
@@ -30,6 +32,69 @@
...
+
+    def bind(self, field_name, parent):
+        """
+        Called when the field is bound to the serializer.
+
+        Changes the source  so that the original field name is used (removes
+        the _id suffix).
+        """
+        if field_name:
+            self.source = field_name[:-len(self.field_name_suffix)]
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: Can you add a comment or docstring like this:
`In case of many=True relations this class is used child_relation class, see first line in many_init() classmethod below. In this case the field_name is an empty string which of course cannot be shortened.`

File: openslides/utils/rest_api.py [hunk 3]
```diff
@@ -30,6 +32,69 @@
...
+
+    Only works together the our ModelSerializer.
+    """
+    field_name_suffix = '_id'
+
+    def bind(self, field_name, parent):
+        """
+        Called when the field is bound to the serializer.
+
+        Changes the source  so that the original field name is used (removes
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: Unnecessary whitespace after `source`.

File: openslides/utils/rest_api.py (hunk 1 above)
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: May be use also a docstring at the bind method, see docstrings in lin 53 ff.

File: openslides/utils/rest_api.py [hunk 4]
```diff
@@ -30,6 +32,69 @@
...
+        """
+        if field_name:
+            self.source = field_name[:-len(self.field_name_suffix)]
+        super().bind(field_name, parent)
+
+    @classmethod
+    def many_init(cls, *args, **kwargs):
+        """
+        Method from rest_framework.relations.RelatedField That uses our
+        IdManyRelatedField class.
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: Add `... instead of rest_framework.relations.ManyRelatedField class`

File: openslides/utils/rest_api.py [hunk 5]
```diff
@@ -30,6 +32,69 @@
...
+        """
+        # TODO: this method can be removed if this is merged and released:
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: The issue and pull request were refused so you can remove this comment and may be add a short explanation, that this code is copied ....

File: openslides/utils/rest_api.py [hunk 6]
```diff
@@ -30,6 +32,69 @@
...
+    """
+    ModelSerializer that changes the field names of related fields to
+    FIELD_NAME_id.
+    """
+    serializer_related_field = IdPrimaryKeyRelatedField
+
+    def get_fields(self):
+        fields = super().get_fields()
+        new_fields = type(fields)()
+        self.manipulated_field_names = set()
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: I do not see/find where this attribute is used? Is this a misstake?

File: openslides/utils/rest_api.py [hunk 7]
```diff
@@ -30,6 +32,69 @@
...
+
+class ModelSerializer(_ModelSerializer):
+    """
+    ModelSerializer that changes the field names of related fields to
+    FIELD_NAME_id.
+    """
+    serializer_related_field = IdPrimaryKeyRelatedField
+
+    def get_fields(self):
+        fields = super().get_fields()
```
Comment (normanjaeckel) [inline openslides/utils/rest_api.py]: Can you please add a docstring and maybe rename new_fields to manipulated_fields or customized_fields or something like that so we don't think these fields are *newley created/initiated* but only some of them are changed/renamed in a special way?

**Comment by normanjaeckel** [comment]
Looks nice (and complex). Some remarks.

File: openslides/utils/rest_api.py (hunk 1 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done

File: openslides/utils/rest_api.py (hunk 2 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done

File: openslides/utils/rest_api.py (hunk 3 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done

File: openslides/utils/rest_api.py (hunk 4 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done

File: openslides/utils/rest_api.py (hunk 5 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done

File: openslides/utils/rest_api.py (hunk 6 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done

File: openslides/utils/rest_api.py (hunk 7 above)
Comment (ostcar) [inline openslides/utils/rest_api.py]: done (in some other way)

**Comment by normanjaeckel** [comment]
Rebase and merge.
